web.browser/src/redux/reducer.js

```diff
@@ -0,0 +1,36 @@
+import { data } from '../mock-data';
+import {
+    SORT_BY_DATE,
+    SORT_BY_UPVOTE,
+    UPDATE_VOTE
+} from './actions';
+
+
+export function PostsReducer(state = data.posts, action) {
+  switch (action.type) {
+    case UPDATE_VOTE:
+        return state.map((post) => {
+            if (action.id === post.id) post.votes++;
+            return post;
+        });
+    case SORT_BY_DATE:
+        let sortedState = [...state]
+        return sortedState.sort((a, b) => {
+            return b.id - a.id;
+        });
+    case SORT_BY_UPVOTE:
+        sortedState = [...state]
+         return sortedState.sort((a, b) => {
+            return b.votes - a.votes;
+        });
```

Comment:
For `SORT_BY_DATE` and `SORT_BY_UPVOTE` you don't need to assign to the `sortedState` var.

```js
case SORT_BY_DATE:
  return [...state].sort((a, b) => {
      return b.id - a.id;
  });
```